src/content/en/updates/2017/12/modulepreload.md

```diff
@@ -0,0 +1,147 @@
+project_path: /web/_project.yaml
+book_path: /web/updates/_book.yaml
+description: &lt;link rel="modulepreload"&gt; offers a way of declaratively loading JavaScript modules ahead of time.
+
+{# wf_updated_on: 2017-12-14 #}
+{# wf_published_on: 2017-12-14 #}
+{# wf_tags: performance, modules #}
+{# wf_blink_components: Blink>Loader #}
+{# wf_featured_image: /web/updates/images/generic/timeline.png #}
+{# wf_featured_snippet: &lt;link rel="modulepreload"&gt; offers a way of declaratively loading JavaScript modules ahead of time. This article looks at how it works and why it's better for modules than &lt;link rel="preload"&gt;. #}
+
+# Preloading modules {: .page-title }
+
+{% include "web/_shared/contributors/sgomes.html" %}
+
+Browsers are finally starting to natively support JavaScript modules, both with
+[static](https://jakearchibald.com/2017/es-modules-in-browsers/) and
+[dynamic](/web/updates/2017/11/dynamic-import)
+import support. This means it's now possible to write module-based
+JavaScript that runs natively in the browser, without transpilers or bundlers.
+
+Module-based development offers some real advantages in terms of cacheability,
+helping you reduce the number of bytes you need to ship to your users.
+The finer granularity of the code also helps with the loading story, by letting you
+prioritise the critical code in your application.
+
+However, module dependencies introduce a loading problem, in that the browser needs
+to wait for a module to load before it finds out what its dependencies are. One way
+around this is by preloading the dependencies, so that the browser knows about all
+the files ahead of time and can keep the connection busy.
+
+Until now, there wasn't really a good way of declaratively preloading modules.
+Chrome 64 ships with `<link rel="modulepreload">` enabled by default, a module-specific
+version of `<link rel="preload">` that solves a number of the latter's problems.
+
+Warning: It's still very much early days for modules in the browser, so while we
+encourage experimentation, we do not recommend using this technology in production
+just yet!
```

Comment:
Why would we discourage using modules in production? :(

Reply:
Even if we're just talking about static imports, it requires significant investment from developers, because they need to both maintain their existing bundled approach for non-module-enabled browsers, as well as create a new output target for module-enabled ones. And even there, it's somewhat easy to shoot yourself in the foot, performance-wise, for now. I'll change the wording to advise caution instead, though.